### PR TITLE
Created MVT returning function for DO

### LINKED ISSUE #333: Create a function to return MVTs from DO
Create a function to return MVTs from DO

### PR DESCRIPTION
**Pull request by antoniocarlon**
Creating an MVT returning function for DO

Closes #333

### CODE REVIEW

**Comment by ethervoid** [comment]
Is this ready for review?

**Comment by antoniocarlon** [comment]
Yes, please CR it!

File: src/pg/sql/45_observatory_mvt.sql [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+      FROM json_to_recordset(meta)
+        AS x(id TEXT, numer_id TEXT, numer_aggregate TEXT, numer_colname TEXT, numer_geomref_colname TEXT, numer_tablename TEXT,
+            numer_type TEXT, denom_id TEXT, denom_aggregate TEXT, denom_colname TEXT, denom_geomref_colname TEXT, denom_tablename TEXT,
+            denom_type TEXT, denom_reltype TEXT, geom_id TEXT, geom_colname TEXT, geom_geomref_colname TEXT, geom_tablename TEXT,
+            geom_type TEXT, numer_timespan TEXT, geom_timespan TEXT, normalization TEXT, api_method TEXT, api_args JSON);
+
+    IF numer_tablenames_do IS NULL OR numer_colnames_do IS NULL OR numer_colnames_do_qualified IS NULL OR numer_colnames_do_normalized IS NULL
+      OR geom_tablenames IS NULL OR geom_colnames IS NULL OR geom_geomref_colnames IS NULL OR geom_geomref_colnames_qualified IS NULL
+      OR geom_relations_do IS NULL THEN
+      RETURN;
```
Comment (ethervoid) [inline src/pg/sql/45_observatory_mvt.sql]: we should provide some kind of information

File: src/pg/sql/45_observatory_mvt.sql [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+    FOREACH numer_tablename_do IN ARRAY numer_tablenames_do LOOP
+      i := i + 1;
+      numer_tablenames_do_outer := numer_tablenames_do_outer || 'LEFT OUTER JOIN ' || numer_tablename_do || ' ON ' || geom_relations_do[i] || ' ';
+    END LOOP;
+  ELSE
+    getmeta_parameters := '[{"geom_id":"' || geography_level || clipped ||'"}]';
+    meta := cdb_observatory.obs_getmeta(geom, getmeta_parameters::json, 1::integer, 1::integer, 1::integer);
+
+    IF meta IS NULL THEN
+      RETURN;
```
Comment (ethervoid) [inline src/pg/sql/45_observatory_mvt.sql]: Same here

File: src/pg/sql/45_observatory_mvt.sql [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+    END IF;
+
+    SELECT  (array_agg(distinct 'observatory.'||geom_tablename))[1] geom_tablenames,
```
Comment (ethervoid) [inline src/pg/sql/45_observatory_mvt.sql]: Here we are only getting the geometries. What is the case? Not passing numer_ids?

File: src/pg/sql/45_observatory_mvt.sql [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+        numer_colnames_mc := coalesce(numer_colnames_mc, '')||numer_colnames_mc_current||',';
+      END IF;
+      IF numer_colnames_mc_qualified_current IS NOT NULL THEN
+        numer_colnames_mc_qualified := coalesce(numer_colnames_mc_qualified, '')||numer_colnames_mc_qualified_current||',';
+      END IF;
+      IF numer_colnames_mc_normalized_current IS NOT NULL THEN
+        numer_colnames_mc_normalized := coalesce(numer_colnames_mc_normalized, '')||numer_colnames_mc_normalized_current||',';
+      END IF;
+
+      IF mc_table IS NOT NULL THEN
```
Comment (ethervoid) [inline src/pg/sql/45_observatory_mvt.sql]: What if mc_table is null?

File: src/pg/sql/45_observatory_mvt.sql [hunk 5]
```diff
@@ -0,0 +1,562 @@
...
+    SELECT  mvtgeom,
+            (select row_to_json(_)::jsonb from (select id, %9$s %3$s area_ratio, area) as _) as mvtdata
+          FROM (
+      SELECT ST_AsMVTGeom(ST_Transform(the_geom, 3857), $1, $2, $3, $4) AS mvtgeom, %8$s as id, %6$s %7$s area_ratio, area FROM (
+        SELECT  %1$s the_geom, %8$s, %2$s %10$s
+                CASE  WHEN ST_Within($5, %1$s)
+                        THEN ST_Area($5) / Nullif(ST_Area(%1$s), 0)
+                      WHEN ST_Within(%1$s, $5)
+                        THEN 1
+                      ELSE ST_Area(cdb_observatory.safe_intersection(st_simplifyvw(%1$s, $6), $5)) / Nullif(ST_Area(%1$s), 0)
```
Comment (ethervoid) [inline src/pg/sql/45_observatory_mvt.sql]: Do we need to use `safe_intersection`? I remember that was outdated by the new intersection from Postgis

File: src/pg/sql/45_observatory_mvt.sql (hunk 4 above)
Comment (antoniocarlon) [reply src/pg/sql/45_observatory_mvt.sql]: In this case, the variables below will have the default value and everything will work as if there's only DO data

File: src/pg/sql/45_observatory_mvt.sql (hunk 3 above)
Comment (antoniocarlon) [reply src/pg/sql/45_observatory_mvt.sql]: `meta` can be null if none of the requested numerators is available. In this case, we only get the geometries

File: src/pg/sql/45_observatory_mvt.sql (hunk 1 above)
Comment (antoniocarlon) [reply src/pg/sql/45_observatory_mvt.sql]: We can take a look at this behaviour in the next iteration if you agree

**Pull request merged**